### ISSUE TITLE
[BUG] Gallery.dll

### ISSUE THREAD

**Issue by NamelessButler** [yotan-dev/mad-island-mods#3]
### Crash - Gallery.dll
Forgot to report before.

### How to reproduce
1. Have to npcs that does not have s#x animation next to each other.
2. Have `relation` and `allow s#x` enabled
3. Have one one of the Leaf Houses build in their territory radius
4. When they try to have s#x, multiple reports of it will be logged, with some exception errors.
5. Game will became very laggy  (when you approach the npcs) until it stops responding and finally crash
Might happen with other npcs. Not fully tested.

### Log
[GalleryLogger.txt](https://github.com/user-attachments/files/17613059/GalleryLogger.txt)

### Game Version
```0.2.1```

PS: You just posted that `v1` mods are for 0.1.8 game version, so you might be aware of this issue, if so feel free to close it.

**Comment by yotan-dev** [comment]
I will take a look.

Only NPC Stats has a differentiation between v0.1.8 and v0.2.1 because I can't have the same dll for both due to some internals.

v0.1.8 uses an `int` for `upAttack` while v0.2.x uses  `float`, during compilation the dll assumes the type from the game assemblies being used at the time, so linking my mod to v0.1.8 assemblies would make it expect an `int` at run time.

The latest version of the other mods are expected to work on both v0.1.8 and v0.2.1 using the same dll, so this is probably a bug

**Comment by yotan-dev** [comment]
weird, I can't seem to reproduce the issue.. tried putting an native girl and native female together, they triggered many times but I did not notice any fps loss. I do have a powerful pc, so this might be preventing me from seeing the issue...

I checked the logs and I can only see issues raising on 22/10, but are for other scenes. note that "S_x Count trace" are not bugs (although they are traces), I just print them like that in case something new shows up I can track what triggered it, but overall the logs seem normal...

Does disabling this mod makes the FPS loss go away? and when you get errors, does it pop up in the middle of the screen with like "Gallery bug report it"?

**Comment by NamelessButler** [comment]
My game is on HDD, I have BepInEx debug console enable so I can track what's going wrong.
What would usually happen is, when I approached the Male/Female old npcs hut,
Game would became extremely laggy, freeze and shortly after crash. meanwhile the console would keep report that
X tried to S_x Y but since no Animation is found skipped.

I tried going to another location, it would appear but to be expected honestly.
But as Soon as I approached the old couple it would go nuts.

I cleared the Log a couple times before, gonna do it again and try to reproduce the issue.
And yeah it only happens with gallery.dll enabled.
